NEXT-12789 - Administration: Handling media (#212)
shopware/docs: guides/plugins/plugins/administration/handling-media.md

# Handling media

> 

Reviewer: Please write a little Overview of the guide as introduction.

## The media upload component

The Shopware 6 Administration media upload component makes it relatively easy to upload media of various kinds such as images, videos and audio files.
This is done through the `sw-media-upload-v2` component as seen below:

```html
<div>
    <sw-media-upload-v2
        uploadTag="my-upload-tag"
        :allowMultiSelect="false"
        variant="regular"

>         :autoUpload="true"
> 		label="My image-upload">

Reviewer 2: ```suggestion
        :autoUpload="true"
        label="My image-upload">
```

    </sw-media-upload-v2>
</div>
```

> As you saw in the code sample, the `sw-media-upload-v2` is pretty configurable trough props.

Reviewer 2: ```suggestion
As you can see in the code sample below, the `sw-media-upload-v2` is pretty configurable through props.
```

just my taste

Reviewer: Should we use the word `properties`, as long version?

| `allowMultiSelect` | Sets whether multiple files can be uploaded at once                                                                             |
| `label`            | The text that is displayed in the header                                                                                        |
| `defaultFolder`    | The path where the file will be put                                                                                             |
| `targetFolderId`   | The `targetFolderId` that will be used as a backup to the `defaultFolder`                                                       |
| `helpText`         | Sets the `helpText` displayed in the header of the component                                                                    |
| `fileAccept`       | Sets what the underlying <input>, accepts standard is `image/*`                                                                 |
| `disabled`         | Disables the whole component                                                                                                    |

## Keeping track of uploads

> The `sw-upload-listener` component, can as seen below, be used in conjunction with an `sw-media-upload-v2` component.

Reviewer: ```suggestion
As seen below, the `sw-upload-listener` component can be used in conjunction with an `sw-media-upload-v2` component.
```

```html
<div>
    <sw-media-upload-v2
        uploadTag="my-upload-tag"
        :allowMultiSelect="false"

>         variant="regular"
> 		label="My image-upload">

Reviewer 2: ```suggestion
        variant="regular"
        label="My image-upload">
```

    </sw-media-upload-v2>
    <sw-upload-listener
        @media-upload-finish="onUploadFinish" 
        uploadTag="my-upload-tag">
    </sw-upload-listener>
</div>
```

Notice that the `uploadTag` needs to be the same in the `sw-media-upload-v2` and the `sw-upload-listener` for them to communicate properly.
Beyond the `media-upload-finish` event there are a few more events:

| Event                 | Description                                        |
|-----------------------|----------------------------------------------------|
| `media-upload-add`    | This event is triggered when an upload is added    |
| `media-upload-finish` | This event is triggered when an upload finishes    |

> | `media-upload-fail`   | This is triggered on an upload failing             |

Reviewer 3: ```suggestion
| `media-upload-fail`   | This event is triggered on an upload failing       |
```

| `media-upload-cancel` | This event is triggered when an upload is canceled |

## Previewing Media

Media can be previewed with the `sw-media-preview-v2` component as seen below:

```html
<sw-media-preview-v2
    :source="some-id">
</sw-media-preview-v2>
```

> As previously mention this component is already embedded within the `sw-media-upload-v2`.

Reviewer 3: ```suggestion
As previously mentioned this component is already embedded within the `sw-media-upload-v2`.
```

> But using it as a separate component you get access to the following configuration options:

Reviewer: ```suggestion
However, using it as a separate component you get access to the following configuration options:
```

> However, using it as a separate component you get access to the following configuration options:

Reviewer 3: ```suggestion
However, using it as a separate component gets you access to the following configuration options:
```

| Property         | Function                                                                            |
|------------------|-------------------------------------------------------------------------------------|
| `source`         | The `id` or alternately the path to the media to be previewed                       |
| `showControls`   | Controls whether media such as videos or audio shows controls                       |
| `autoplay`       | Controls whether media such as videos or audio autoplays                            |

> | `hideTooltip`    | Hides the tooltip witch the filename of the media in at the bottom of the component |

Reviewer 3: ```suggestion
| `hideTooltip`    | Hides the the filename tooltip of the media in at the bottom of the component       |
```
